static/js/copy_and_paste.js

```diff
@@ -306,6 +306,20 @@ export function paste_handler_converter(paste_html) {
     return markdown_text;
 }
 
+// https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url
+function validURL(str) {
+    const pattern = new RegExp(
+        "^(https?:\\/\\/)?" + // protocol
+            "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // domain name
+            "((\\d{1,3}\\.){3}\\d{1,3}))" + // OR ip (v4) address
+            "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // port and path
+            "(\\?[;&a-z\\d%_.~+=-]*)?" + // query string
+            "(\\#[-a-z\\d_]*)?$",
+        "i",
+    ); // fragment locator
+    return Boolean(pattern.test(str));
```

Comment:
Surely there's a standard library function we can use here.

Reply:
What standard library are you talking about here?

Reply:
I changed it to `URL(str)`, check if this feels right.

Reply:
Yeah, this looks right; you can also drop the comment since we don't need a reference to see that `new URL` is correct.